### ISSUE TITLE
Require either `async override` or `override async` (don't allow both)

### ISSUE THREAD

**Issue by Skalman** [microsoft/TypeScript#43606]
# Bug Report

### 🔎 Search Terms

class async override

### 🕗 Version & Regression Information

- This changed between versions __4.2.0__ and __nightly__

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.3.0-dev.20210408&ssl=1&ssc=1&pln=2&pc=1#code/MYGwhgzhAEBCkFNoG8CwAoaXoFsEBcALAewBMBGACgEoUBfDbXAk0gJhvowfQ1EhgAVBBHzQEAD3wIAdqRjwISNJmwAHAK4AjEAEtg0YgDcEAJ1O7SSSAE8ZBvETJVaKpk1MENpmdAAKpsQ4ukoAdJ4QxCAmlOTUANyM2DxJWJo6+tC29oYm5pZIjqwcrqnunvjevgFBIQjhIlExcYmqWDw8QA)

### 💻 Code

```ts
class Base {
    method1() {}
    method2() {}
}

class Test extends Base {
    public override async method1() {
        return Promise.resolve(1);
    }

    public async override method2() {
        return Promise.resolve(1);
    }
}
```

### 🙁 Actual behavior

No errors

### 🙂 Expected behavior

Possibly error on either `override async` or `async override`.

My understanding (from #43533) is that you prefer requiring a specific modifier order. If that's the case, I would have expected an error here.

**Comment by dsherret** [comment]
For some more information on which one should be chosen if one should be chosen, C# supports both forms, but `override async` is used about 12x more than `async override`. Also, doing `override async` would keep in line with having the `readonly` keyword appear after `override` as mentioned in that linked issue, Additionally, `override async` is winning big time in [this poll](https://twitter.com/DavidSherret/status/1380331297931943940) I did.

Right now, it seems like the `declare` keyword is the only modifier whose order isn't enforced.

**Comment by fbartho** [comment]
just like `public/private`, `override` is something of an internal annotation (for implementors), while `async` is a more external annotation (for consumers).

Therefore, it makes sense to me to spell it as `public override readonly async someMethod()` which follows a consistent direction of moving along a spectrum from implementor-relevant towards consumer-focused (left-to-right).

**Comment by fbartho** [comment]
To put it another way:
- `async` applies to the right to `someMethod()`
- readonly applies to the instance (what you can do to an instance)
- override and public both apply outwards to the class or meta level

(Not sure if this phrasing makes more sense to people)

**Comment by andrewbranch** [comment]
The order of `override` and `static` is also not currently enforced.
